api/src/main/java/org/pmiops/workbench/opsgenie/EgressEventServiceImpl.java

```diff
@@ -0,0 +1,200 @@
+    request.setAlias(egressEvent.getProjectName() + " | " + egressEvent.getVmName());
+    return request;
+  }
+
+  @NotNull
+  private String getDescription(EgressEvent egressEvent) {
+    final WorkspaceAdminView adminWorkspace =
+        workspaceAdminService.getWorkspaceAdminView(egressEvent.getProjectName());
+    final Workspace workspace = adminWorkspace.getWorkspace();
+    final String creatorDetails =
+        userService
+            .getByUsername(workspace.getCreator())
+            .map(this::getAdminDescription)
+            .orElse("Creator not Found");
+
+    final Optional<DbUser> executor =
+        vmNameToUserDatabaseId(egressEvent.getVmName()).flatMap(userService::getByDatabaseId);
+
+    final String executorDetails =
+        executor.map(this::getAdminDescription).orElse("Executing User not Found");
+
+    final String collaboratorDetails =
+        adminWorkspace.getCollaborators().stream()
+            .map(this::formatWorkspaceUserAdminView)
+            .collect(Collectors.joining("\n\n"));
+
+    return String.format(
+            "Workspace \"%s\", Age = %d Days\n",
+            workspace.getName(),
+            getAgeInDays(Timestamp.from(Instant.ofEpochMilli(workspace.getCreationTime()))))
+        + String.format(
+            "GCP Billing Project/Firecloud Namespace: %s\n", egressEvent.getProjectName())
+        + String.format("Notebook Jupyter VM name: %s\n", egressEvent.getVmName())
+        + String.format("MySQL workspace_id: %d\n", adminWorkspace.getWorkspaceDatabaseId())
+        + String.format(
+            "Egress detected: %.2f Mib in %d secs\n\n",
+            egressEvent.getEgressMib(), egressEvent.getTimeWindowDuration())
+        + String.format(
+            "Cluster Name: %s\n", executor.map(DbUser::getClusterName).orElse("unknown"))
+        + String.format("User Running Notebook: %s\n\n", executorDetails)
+        + String.format("Workspace Creator: %s\n\n", creatorDetails)
+        + String.format("Collaborators: \n%s\n", collaboratorDetails)
+        + String.format(
+            "Workspace Admin Console (Prod Admin User): %s/admin/workspaces/%s/\n",
+            workbenchConfigProvider.get().server.uiBaseUrl, egressEvent.getProjectName())
+        + "Playbook Entry: https://broad.io/aou-high-egress-event";
+  }
+
+  private String formatWorkspaceUserAdminView(WorkspaceUserAdminView userAdminView) {
+    final String userDetails =
+        userService
+            .getByDatabaseId(userAdminView.getUserDatabaseId())
+            .map(this::getAdminDescription)
+            .orElse(
+                String.format(
+                    "Collaborator with user_id %d not Found", userAdminView.getUserDatabaseId()));
+    return String.format("%s: %s", userAdminView.getRole(), userDetails);
+  }
+
+  /**
+   * Produce a succinct string for giving an administrator a heads-up in an alert or other text-only
+   * context.
+   */
+  @Transient
+  public String getAdminDescription(DbUser dbUser) {
```

Comment:
@calbach per our discussion, I've removed the contact email. I believe the remainder of the fields are in the researcher directory (name & institution).

I wonder if we need a global identifier for users that's completely anonymous for some purposes.

Reply:
Thanks.